rpc/rpc-transport/socket/src/socket.c

```diff
@@ -4161,6 +4161,8 @@ ssl_setup_connection_params(rpc_transport_t *this)
             }
         }
 
+/* Only needed on older than 3.0 OpenSSL. On newer one, use default values. */
+#if OPENSSL_VERSION_NUMBER < 0x030000000  // 3.0.0
```

Comment:
Even if default values from 3.0 are better, they could eventually become non-optimal, as it happened with previous versions. Shouldn't we keep the possibility to manually modify the defaults ?

Reply:
That's what the suggestion wherever I read was - don't change. I'm open for alternatives.

Reply:
I don't have a strong opinion here. I just thought that keeping the options to modify them will add flexibility in some cases (like algorithm incompatibility or potential weaknesses detected in the future). However, if it's recommended to not touch them, I'm also ok. As you prefer.